directory/directory_transport.go

```diff
@@ -157,7 +157,11 @@ func (ref dirReference) NewImageDestination(ctx context.Context, sys *types.Syst
 	if sys != nil {
 		compress = sys.DirForceCompress
 	}
-	return newImageDestination(ref, compress)
+	decompress := false
+	if sys != nil {
+		decompress = sys.DirForceDecompress
+	}
+	return newImageDestination(ref, compress, decompress)
```

Comment:
`compress && decompress` should fail with an error, compare https://github.com/containers/skopeo/pull/1365#discussion_r665677728 .